File: backend/src/main/java/com/backend/global/security/CustomUserDetails.java
/*
package com.backend.global.security;

import com.backend.domain.member.entity.Member;
import lombok.Getter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.Collections;

@Getter
public class CustomUserDetails implements UserDetails {

    private final Long id;           // Member PK
    private final String memberId;   // 로그인 ID
    private final String password;
    private final String nickname;
    private final String role;

    public CustomUserDetails(Member member) {
        this.id = member.getId();
        this.memberId = member.getMemberId();
        this.password = member.getPassword();
        this.nickname = member.getNickname();
        this.role = member.getRole().name();
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return Collections.singleton(() -> "ROLE_" + role);
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public String getUsername() {
        return memberId; // 로그인용 ID 기준
    }

    @Override
    public boolean isAccountNonExpired() { return true; }

    @Override
    public boolean isAccountNonLocked() { return true; }

    @Override
    public boolean isCredentialsNonExpired() { return true; }

    @Override
    public boolean isEnabled() { return true; }
}
*/

File: backend/src/main/java/com/backend/global/security/user/CustomUserDetails.kt
package com.backend.global.security.user

import com.backend.domain.member.entity.Member
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class CustomUserDetails(member: Member) : UserDetails {

    val id: Long? = member.id
    val memberId: String = member.memberId
    private val password: String = member.password
    val nickname: String = member.nickname
    private val role: String = member.role.name

    override fun getAuthorities(): MutableCollection<out GrantedAuthority> =
        mutableSetOf(GrantedAuthority { "ROLE_$role" })

    override fun getPassword() = password

    override fun getUsername() = memberId // 로그인용 ID 기준

    override fun isAccountNonExpired() = true

    override fun isAccountNonLocked() = true

    override fun isCredentialsNonExpired() = true

    override fun isEnabled() = true
}
